File: chrome/test/data/webui/chrome_timeticks_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/test/browser_test.h"

using ChromeTimeTicksBrowserTest = WebUIMochaBrowserTest;

IN_PROC_BROWSER_TEST_F(ChromeTimeTicksBrowserTest, All) {
  // Spawn the test from chrome://settings (or any WebUI surface) so that
  // the chrome.timeTicks API exists.
  set_test_loader_host(chrome::kChromeUISettingsHost);
  RunTest("chrome_timeticks_test.js", "mocha.run()");
}

File: chrome/test/data/webui/chrome_timeticks_browsertest.js
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview Tests to ensure that chrome.timeticks.nowInMicroseconds()
 * returns a BitInt.
 */

GEN('#include "content/public/test/browser_test.h"');

/** Test fixture for chrome.timeTicks WebUI testing. */
var ChromeTimeTicksBrowserTest = class extends testing.Test {
  /** @override */
  get browsePreload() {
    return 'chrome://webui-test/test_loader.html?module=chrome_timeticks_test.js';
  }

  /** @override */
  get isAsync() {
    return true;
  }

  /** @override */
  get webuiHost() {
    return 'dummyurl';
  }
};

TEST_F('ChromeTimeTicksBrowserTest', 'All', function() {
  mocha.run();
});
